src/types/jsx.d.ts

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import 'react'
+
+declare module 'react' {
```

Comment:
See https://react.dev/blog/2024/04/25/react-19-upgrade-guide#the-jsx-namespace-in-typescript for more details.